### ISSUE TITLE
2015-07-25 International NodeBots Day

### ISSUE THREAD

**Issue by spikeheap** [jsoxford/jsoxford.github.com#103]
NodeBots are robots that are controlled by node.js. A NodeBots event is a full day event where JS developers team up in small teams to program robots with JS.

We'll have arduino kits you can use to build and program robots. You don't need to know what a resistor is or how to use an aynschronous callback, we'll have hardware and JS experts ready to help you throughout the day. We've also got 3 radio-controlled, video-enabled cars which can be controlled with JS over wifi, and with  extra sensors, displays and buckets of Lego Technic, it'll be a fun day of prototyping and hacking.

You'll be able to buy Arduino kits at a discount on the day, if you want to take your work home with you (sorry you can't buy the lego though!)

You will need to bring your own laptop, and it will help if you have already installed node.js. We'll have plenty of mentors to help and guide you throughout the day.

#### Date
Saturday 25th July 2015, 10:00am

#### Organisation

- [ ] Raise invoice with Haybrook for sponsorship (@spikeheap has the figures)
- [x] Structure? #56 @AverageMarcus had some good points on this.
-  ~~Nodecopter?~~
-  ~~Untethered Nodebots? #40 (:+1:)~~
- [x] Order/procure hardware kits/parts (see #121)
- [x] Talk to Espruino about sponsorship/attendance? Gordon is really keen for this :+1:.
- [x] Book pastries (£45 of small pastries - 9.30 at Ricks)
- [x] Purchase snacks/drinks (see #101)

Part of the Summer of Hacks 2015 (#101).

**Comment by spikeheap** [comment]
I had a great meeting with @gfwilliams this evening and we went through what we'll need to pick up to build the cars as well as approaches we can take. 

To keep it hardware focussed, we'll start from scratch and let attendees solder everything onto the boards and build the car up from components. That shouldn't take more than the morning session (at a stretch), so everyone should be caught up before the afternoon session. We need to knock up some basic instructions and ideas for goals, but my current thinking is:

0. Play with the Espruino & web IDE (get the light to flash, etc.).
1. Build the robot, soldering and bluetacking bits together.
2. Get the car to move forward, backward, and turn 90(ish) degrees.
3. Light up the LEDS and read from the light sensors mounted on the front.
4. Follow a line somewhere (sharpies on paper) using (3). This is a big step!
5. Read from the optical sensor and try to follow a line (or do something cool) with that. This is probably a bit of a stretch!

Gordon suggested adding some breakout pins onto the board layout so if people are keen to add weird and wonderful additions it should be possible. 

So for the event we'll need:

* Sharpies.
* A4 (or A3) white paper.
* Bluetack.
* Hot glue guns (@gfwilliams has a couple, but if anyone else has one they could bring that would be ace).
* Soldering irons (we have 5-6 between @gfwilliams, @spikeheap, @peterjwest and @AverageMarcus, again additional irons would probably help).
* USB extension leads. Ideally 15 (1 per group), but just as many as we can scrounge.

The servos we're using for wheels pull the car along nicely. The servos come with a small plastic disc, but that lacks traction so we need to fashion wheels, something like:

* Champagne (or prosecco @benfoxall) corks, as long as they're >24mm.
* Beer bottle tops >24mm (spikey but should get grip).
* Water bottle tops (>24mm), wrapped in elastic bands (<24mm)

#### Actions

- [ ] Sort out grippy wheels x30.
- [ ] Write tutorial bits.

**Comment by AverageMarcus** [comment]
@spikeheap one of my soldering irons is gas powered, not sure how WO would feel about me using it there.

**Comment by spikeheap** [comment]
@AverageMarcus that's a good point in general. @benfoxall/@peterjwest could we check if there are any issues around soldering irons and that sort of thing. Do we need a disclaimer/waiver or something like that?

**Comment by AverageMarcus** [comment]
@spikeheap I've ordered myself a glue gun this morning (been meaning to for ages now) and I'm fairly sure I have a pretty long USB extension cable so that's at least 1 groups covered. I'll prob be bringing my espruinos and various bits long too.

**Comment by spikeheap** [comment]
@AverageMarcus :+1: :+1:.

**Comment by gfwilliams** [comment]
Hi,

I've actually got 4 hot glue guns and a bunch of glue sticks (don't ask) so I think on that front we're probably sorted. I could also bring 2 soldering irons. 

Personally I wouldn't bring any gas ones - as @AverageMarcus says there could be some fire issues that you wouldn't have with electric ones - also while they're fine for doing wire, I reckon you'd make a bit of a mess of a PCB with one?

I guess other things that'd be needed are:

* Side cutters - to trim component legs once soldered (I could bring 2 of those)
* Solder
* Work Clamps - a kind of [helping hand](http://www.amazon.co.uk/Helping-Magnifier-Strong-Alligator-Crocodile/dp/B004VMV6D8) thing. While not vital they do make soldering some things easier. I can bring 1 of those.

For the wheels, what about [something like this?](http://uk.essentracomponents.com/shop/en-GB/essentracomponentsuk/feet--casters---glides-29054-44/self-adhesive-bumpers---rubber-feet-348552-44/self-adhesive-bumpers---rubber-feet-348552-44/round-self-adhesive-bumper---rubber-foot%3A-clear--457-mm-base-diameter--152-mm-base-height-1011630) or [Sod138 from here](http://www.componentforce.co.uk/category/586/stick-on-discs)?

Not sure what they'd be like, but it could be a nice easy way to get a little bit more grip while keeping them all the same.

**Comment by gfwilliams** [comment]
Sorry for the spam, but I think I just found a good solution for the wheels: [lobster bands!](http://www.gtproductsmarine.com/shop/lobster-band-rubber-25mm-x-13mm-red.html)

Just checking if 25mm is the circumference (looks like it) - if it is they'll be perfect to stretch over the existing bit on the servo.

**Comment by peterjwest** [comment]
I think we'll be fine using electrical soldering irons in the office, we could possibly do with something heat proof to put them on.

I don't really think we need a disclaimer, but if someone wants to organise one, then go ahead.

**Comment by spikeheap** [comment]
@gfwilliams that was a great call with the lobster bands. Of course I ordered them in yellow ;):
![img_0244](https://cloud.githubusercontent.com/assets/464482/8429224/8c7d6ffe-1f1d-11e5-9cb2-454a865ae488.JPG)

**Comment by spikeheap** [comment]
Right, I've ordered the outstanding parts, so we're all set for next Saturday 🔥. 

I *haven't* ordered bluetooth or WiFi modules because at £4-5 each when ordered from the UK it bumped the price up too much. I think we'll have easily enough to keep people entertained for the day without it, and we can post on the Meetup to let people know what to but if they want to contribute a board...

So... I think the only other things we need to bring on the day now are:

- [ ] borrowed soldering irons (as many as possible)
- [ ] battery chargers
- [ ] multimeters
- [ ] helping hands claw things (if anyone has any kicking around)

Basically, if it's related and you have it, we could probably do with more of them!

Prior to the event we need:

- [ ] Some ideas/simple steps to work through. @gfwilliams created [this](https://github.com/espruino/EspruinoDocs/blob/master/devices/RobotPCB.md) page for the board (which is awesome) 
- [ ] To build up one of the cars, taking a few pictures and checking against the docs
- [ ] Tweet!
- [ ] Message attendees about the optional extra modules

**Comment by gfwilliams** [comment]
I'll could bring some WiFi and bluetooth modules if you want - I have a few left over from the KickStarter. I just looked over what they cost me and I could do them for BT £4, WiFi £3 - we could just see if they get used or not?

Also, I've got some other stuff (ultrasonic sensors, LCDs, radio transmitters/receivers, etc) that I could bring just in case, and we just see if stuff gets used or not? It's all relatively cheap (in the £1-3 range) - and as long as it's not bent/soldered to I can just take it back?

I'll bring the LiPo chargers I have - it'd be worth getting some batteries soldered and on charge first thing in the morning.

Also, if anyone has these, please could they bring them?

* tweezers / needle nose pliers
* solder suckers / solder wick

It'd probably be worth doing a demo of soldering the SMD resistors/capacitors beforehand. It's actually pretty easy:

* blob of solder on one pad
* hold resistor in place at one end and melt the blob of solder with the iron
* solder the second pad

but if you don't hold the resistor at first you can get into a hell of a mess :)

**Comment by spikeheap** [comment]
Hi @gfwilliams that's a great idea for the WiFi/BT modules. I've soldered up the batteries so we can get them charged as soon as possible. They all had decent voltages when I checked them, but no idea how long they'll last.

You're spot on about the demo at the beginning. Some attendees may have never touched a soldering iron before, so it'll help to make it feel less intimidating :+1:.

**Comment by spikeheap** [comment]
NodeBots was awesome. Huge thanks to everyone that helped and attended. Rob wrote up the event for our blog: https://gist.github.com/robshep/88e266cbe9c813292567.

**Comment by spikeheap** [comment]
I really enjoyed NodeBots. Maybe we should run another one before next summer?!?

**Comment by AverageMarcus** [comment]
YES!

**Comment by gfwilliams** [comment]
I'd definitely be up for something. It's probably worth not doing something Espruino-specific this time though :)

While it's not robots, something I've fancied for a while is a remote sensor hack day... WiFi on embedded always feels like a fragmented mess, so we could have a bunch of those cheap 433Mhz radios and could set up a Pi that listened for radio messages and did something with them (MQTT server? Maybe also display them on a screen).

Then you could have some people connecting transmitters to their microcontroller of choice (it's literally 3 wires) and transmitting data from whatever sensors they can hook up, then others hacking up web services that did stuff with it.
